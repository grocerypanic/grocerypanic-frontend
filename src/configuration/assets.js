const Assets = {
  favicon: `${process.env.PUBLIC_URL}/favicon.ico`,
  nonBreakingSpace: "\u00A0",
  cart: `${process.env.PUBLIC_URL}/images/cart_desktop.jpg`,
  bill: `${process.env.PUBLIC_URL}/images/bill_desktop.jpg`,
  cart_mobile: `${process.env.PUBLIC_URL}/images/cart_mobile.jpg`,
  bill_mobile: `${process.env.PUBLIC_URL}/images/bill_mobile.jpg`,
};

export default Assets;
