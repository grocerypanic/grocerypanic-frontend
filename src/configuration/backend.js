export const SafeMethods = ["GET", "HEAD", "OPTIONS", "TRACE"];
export const ItemFilters = ["preferred_stores", "shelf"];
export const FilterTag = "filter_tag";

export const Constants = {
  csrfLocalStorage: "csrf-token",
  csrfHeaderName: "X-CSRFToken",
  genericAPIError: "API Error",
  csrfErrorMessage: "Refresh csrf and try again.",
  alreadyRegistered: "User is already registered with this e-mail address.",
  minimumQuanity: 0,
  maximumQuantity: 10000,
  minimumShelfLife: 1,
  maximumShelfLife: 365 * 3,
  defaultShelfLife: 7,
  minimumPrice: 0,
  maximumPrice: 9999,
  maximumTransactionHistory: 14,
  AnalyticsCookieName: "PanicDataConsent-" + process.env.NODE_ENV.toUpperCase(),
  duplicateObjectApiError: { non_field_errors: ["Object already exists."] },
  pageLookupParam: "page",
  pageOverrideParam: "all_results",
};

export const Providers = {
  google: "google",
  facebook: "facebook",
};

export const Paths = {
  googleLogin: "/api/v1/auth/social/google/",
  facebookLogin: "/api/v1/auth/social/facebook/",
  manageShelves: "/api/v1/shelves/",
  manageStores: "/api/v1/stores/",
  manageItems: "/api/v1/items/",
  refreshCSRF: "/api/v1/auth/csrf/",
  manageTransactions: "/api/v1/transactions/",
};

const month = 30;
const year = 365;

export const ShelfLifeConstants = [
  { name: "3 Days", id: 3 },
  { name: "1 Week", id: 7 },
  { name: "2 Weeks", id: 14 },
  { name: "3 Weeks", id: 21 },
  { name: "1 Month", id: month },
  { name: "2 Months", id: month * 2 },
  { name: "3 Months", id: month * 3 },
  { name: "6 Months", id: month * 6 },
  { name: "1 Year", id: year },
  { name: "2 Years", id: year * 2 },
  { name: "3 Years", id: year * 3 },
];
