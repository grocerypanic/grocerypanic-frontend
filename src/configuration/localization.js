import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { Strings } from "./strings";

const i18nConfig = {
  fallbackLng: {
    default: ["en"],
  },
  debug: true,
  ns: ["translation"],
  defaultNS: "translation",
  keySeparator: ".",
  interpolation: {
    escapeValue: false,
    formatSeparator: ",",
  },
  react: {
    wait: true,
  },
};

i18n.use(LanguageDetector).init({
  resources: {
    ...Strings,
  },
  ...i18nConfig,
});

export default i18n;
