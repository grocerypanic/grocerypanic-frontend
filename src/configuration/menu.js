import Routes from "./routes";

const Options = [
  { name: "All Items", location: Routes.items },
  { name: "Shelves", location: Routes.shelves },
  { name: "Stores", location: Routes.stores },
];

export default Options;
