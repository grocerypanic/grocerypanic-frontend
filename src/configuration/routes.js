const Routes = {
  root: "/",
  menu: "/menu",
  about: "/about",
  privacy: "/privacy",
  signin: "/signin",
  shelves: "/shelves",
  splash: "/",
  stores: "/stores",
  items: "/items",
  create: "/create",
  details: "/details/:id",
  goBack: "back",
};

export const External = {
  feedback: "https://www.sharedvisionsolutions.com/#contact",
  credit: "https://www.sharedvisionsolutions.com/",
};

export default Routes;
