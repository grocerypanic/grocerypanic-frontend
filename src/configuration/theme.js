export const primary = "#006600";
export const white = "#FFFFFF";
export const secondary = "#868e96";
export const tertiary = "#808080";
export const error = "#FF0000";
export const regular = "#000000";
export const success = "#5cb85c";
export const highlight = "blue";
export const selected = "red";
export const danger = "#d9534f";
export const dark = "#495057";
export const black = "#000000";
export const warning = "#ffc107";

export const ui = {
  alertTimeout: 2000,
  resizeTimeout: 1,
};

export const navbarSize = "50px";
export const tabSize = "26px";
export const itemAttributes = {
  height: 55,
  padding: 5,
  border: 10,
  spacing: 5,
  innerHeight: 40,
  placeHolderPadding: 12,
};

export const headerMobileThreshold = 480;
export const mobileThreshold = 480;

export const graph = {
  quantityLine: "rgb(2, 117, 216)",
  changeLine: "rgb(92, 184, 92)",
  topPadding: -12,
  holdingPatternHeight: "134px",
  chartContainerHeight: "150px",
  holdingPatternColor: "secondary",
  holdingPatternScale: 1,
  holdingPatternDivHeight: 20,
  holdingPatternAnimation: "grow",
};
