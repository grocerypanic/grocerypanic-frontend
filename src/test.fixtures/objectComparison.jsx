export const propCount = (hash, length) => {
  expect(Object.keys(hash).length).toBe(length);
};
