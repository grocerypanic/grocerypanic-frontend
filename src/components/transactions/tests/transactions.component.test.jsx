import React from "react";
import "jest-canvas-mock";
import { render, cleanup, waitFor } from "@testing-library/react";
import moment from "moment";

import HoldingPattern from "../../holding-pattern/holding-pattern.component";
import TransactionsReview from "../transactions.component";

import { AnalyticsContext } from "../../../providers/analytics/analytics.provider";
import InitialState from "../../../providers/api/transaction/transaction.initial";
import ItemInitialState from "../../../providers/api/item/item.initial";

import Strings from "../../../configuration/strings";
import { generateConverter } from "../../../providers/api/api.util";
import { graph } from "../../../configuration/theme";
import { propCount } from "../../../test.fixtures/objectComparison";

global.Chart = jest.fn();
global.Chart.mockImplementation(() => {
  return {
    render: jest.fn(),
    destroy: jest.fn(),
  };
});
jest.mock("../../holding-pattern/holding-pattern.component");
HoldingPattern.mockImplementation(({ children }) => children);

const mockEvent = jest.fn();
const mockAnalyticsSettings = { event: mockEvent, initialized: true };

// Freeze Time
const OriginalDate = Date.now;
const mockDate = () =>
  (Date.now = jest.fn(() => new Date("2020-06-16T11:01:58.135Z")));

const convertDatesToLocal = generateConverter(InitialState.class);
const convertItemDatesToLocal = generateConverter(ItemInitialState.class);

const mockItem = convertItemDatesToLocal({
  expired: 0,
  id: 1,
  name: "Vegan Cheese",
  next_expiry_date: "2020-06-18",
  next_expiry_quantity: 1,
  preferred_stores: [1],
  price: "4.00",
  quantity: 1,
  shelf: 2,
  shelf_life: 5,
});

const mockTransactions = [
  { id: 1, item: 1, datetime: "2019-09-15", quantity: -5 },
  { id: 2, item: 1, datetime: "2019-10-15", quantity: 5 },
  { id: 3, item: 1, datetime: "2019-11-15", quantity: -5 },
  { id: 4, item: 1, datetime: "2019-12-15", quantity: 5 },
  { id: 5, item: 1, datetime: "2020-01-15", quantity: -5 },
  { id: 6, item: 1, datetime: "2020-01-16", quantity: -5 },
  { id: 7, item: 1, datetime: "2020-02-15", quantity: 5 },
  { id: 8, item: 1, datetime: "2020-03-15", quantity: 5 },
  { id: 9, item: 1, datetime: "2020-06-05", quantity: -1 },
  { id: 10, item: 1, datetime: "2020-06-10", quantity: -3 },
  { id: 11, item: 1, datetime: "2020-06-15", quantity: -3 },
].map((o) => convertDatesToLocal(o));

describe("Setup Environment", () => {
  let utils;
  let current;

  beforeEach(() => {
    mockDate();
    current = {
      item: { ...mockItem },
      ready: false,
      tr: [...mockTransactions],
    };
  });

  const renderHelper = (config) => {
    return render(
      <AnalyticsContext.Provider value={mockAnalyticsSettings}>
        <TransactionsReview {...config} />
      </AnalyticsContext.Provider>
    );
  };

  describe("after transactions are fetched", () => {
    beforeEach(() => {
      current.ready = true;
      jest.clearAllMocks();
    });

    afterEach(cleanup);

    describe("with items in the inventory", () => {
      beforeEach(() => {
        jest.clearAllMocks();
        utils = renderHelper(current);
      });

      it("should display weekly consumption data", () => {
        expect(
          utils.queryByText(`${Strings.ItemStats.ConsumptionConsumedLastWeek}`)
        ).toBeTruthy();
        expect(utils.getByTestId("lastWeek").textContent).toBe("6");
      });

      it("should display monthly consumption data", () => {
        expect(
          utils.queryByText(`${Strings.ItemStats.ConsumptionConsumedLastMonth}`)
        ).toBeTruthy();
        expect(utils.getByTestId("lastMonth").textContent).toBe("7");
      });

      it("should display weekly consumption data", () => {
        expect(
          utils.queryByText(`${Strings.ItemStats.ConsumptionAvgWeek}`)
        ).toBeTruthy();
        expect(utils.getByTestId("avgWeek").textContent).toBe("4.5");
      });

      it("should display weekly consumption data", () => {
        expect(
          utils.queryByText(`${Strings.ItemStats.ConsumptionAvgMonth}`)
        ).toBeTruthy();
        expect(utils.getByTestId("avgMonth").textContent).toBe("6.8");
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should render the graph", async () => {
        await waitFor(() => expect(global.Chart).toBeCalledTimes(1));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });

    describe("with no transactions", () => {
      beforeEach(() => {
        current.tr = [];
        jest.clearAllMocks();
        utils = renderHelper(current);
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should display the insufficient data warning", () => {
        expect(
          utils.queryByText(`${Strings.ItemStats.NotEnoughData}`)
        ).toBeTruthy();
        expect(global.Chart).toBeCalledTimes(0);
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });

    describe("with expired items in inventory", () => {
      beforeEach(() => {
        current.item.expired = 9;
        jest.clearAllMocks();
        utils = renderHelper(current);
      });

      it("should render a warning about the expired items", () => {
        expect(utils.getByText(Strings.ItemStats.RecommendExpiredItems));
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should render the graph", async () => {
        await waitFor(() => expect(global.Chart).toBeCalledTimes(1));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });

    describe("with items expirying soon in inventory", () => {
      beforeEach(() => {
        utils = renderHelper(current);
      });

      it("should render a warning items expiring soon", () => {
        expect(
          utils.getByText(
            `${current.item.next_expiry_quantity} ${Strings.ItemStats.RecommendExpiringSoon}`
          )
        ).toBeTruthy();
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should render the graph", async () => {
        await waitFor(() => expect(global.Chart).toBeCalledTimes(1));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });

    describe("with no items expirying soon in inventory, but an old next_expiry_date set", () => {
      beforeEach(() => {
        Date.now = OriginalDate;
        current.item.next_expiry_date = moment("2019-06-16");
        current.next_expiry_quantity = 0;
        mockDate();
        utils = renderHelper(current);
      });

      it("should not render a warning items expiring soon", () => {
        expect(
          utils.queryByText(
            `${current.item.next_expiry_quantity} ${Strings.ItemStats.RecommendExpiringSoon}`
          )
        ).toBeFalsy();
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(false);
        expect(call.color).toBe("secondary");
        expect(call.animation).toBe("grow");
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(1);
        expect(call.divHeight).toBe(20);
        expect(call.children).toBeTruthy();
      });

      it("should render the graph", async () => {
        await waitFor(() => expect(global.Chart).toBeCalledTimes(1));
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });
  });

  describe("before transactions are fetched", () => {
    beforeEach(() => {
      current.ready = false;
      jest.clearAllMocks();
    });

    describe("with items in the inventory", () => {
      beforeEach(() => {
        jest.clearAllMocks();
        utils = renderHelper(current);
      });

      it("should call the holding pattern with expected arguments", async () => {
        await waitFor(() => expect(HoldingPattern).toBeCalledTimes(1));
        const call = HoldingPattern.mock.calls[0][0];
        propCount(call, 7);
        expect(call.condition).toBe(true);
        expect(call.color).toBe(graph.holdingPatternColor);
        expect(call.animation).toBe(graph.holdingPatternAnimation);
        expect(call.height).toBe(graph.holdingPatternHeight);
        expect(call.scale).toBe(graph.holdingPatternScale);
        expect(call.divHeight).toBe(graph.holdingPatternDivHeight);
        expect(call.children).toBeTruthy();
      });

      it("should not render the graph", async () => {
        expect(global.Chart).toBeCalledTimes(0);
      });

      it("should match the snapshot on file (styles)", () => {
        expect(utils.container).toMatchSnapshot();
      });
    });
  });
});
