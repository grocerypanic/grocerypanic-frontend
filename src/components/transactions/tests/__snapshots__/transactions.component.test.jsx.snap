// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment after transactions are fetched with expired items in inventory should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Consumption Statistics
    </div>
    <div
      class="c1 alert alert-danger mb-2"
    >
      Expired Items Warning
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                last week
              </td>
              <td
                data-testid="lastWeek"
              >
                6
              </td>
            </tr>
            <tr>
              <td>
                last month
              </td>
              <td
                data-testid="lastMonth"
              >
                7
              </td>
            </tr>
            <tr>
              <td>
                avg. per week
              </td>
              <td
                data-testid="avgWeek"
              >
                4.4
              </td>
            </tr>
            <tr>
              <td>
                avg. per month
              </td>
              <td
                data-testid="avgMonth"
              >
                5.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment after transactions are fetched with items expirying soon in inventory should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Consumption Statistics
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                last week
              </td>
              <td
                data-testid="lastWeek"
              >
                6
              </td>
            </tr>
            <tr>
              <td>
                last month
              </td>
              <td
                data-testid="lastMonth"
              >
                7
              </td>
            </tr>
            <tr>
              <td>
                avg. per week
              </td>
              <td
                data-testid="avgWeek"
              >
                4.4
              </td>
            </tr>
            <tr>
              <td>
                avg. per month
              </td>
              <td
                data-testid="avgMonth"
              >
                5.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment after transactions are fetched with items in the inventory should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Consumption Statistics
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                last week
              </td>
              <td
                data-testid="lastWeek"
              >
                6
              </td>
            </tr>
            <tr>
              <td>
                last month
              </td>
              <td
                data-testid="lastMonth"
              >
                7
              </td>
            </tr>
            <tr>
              <td>
                avg. per week
              </td>
              <td
                data-testid="avgWeek"
              >
                4.4
              </td>
            </tr>
            <tr>
              <td>
                avg. per month
              </td>
              <td
                data-testid="avgMonth"
              >
                5.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment after transactions are fetched with no items expirying soon in inventory, but an old next_expiry_date set should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Consumption Statistics
    </div>
    <div
      class="c1 alert alert-danger mb-2"
    >
      Expired Items Warning
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                last week
              </td>
              <td
                data-testid="lastWeek"
              >
                6
              </td>
            </tr>
            <tr>
              <td>
                last month
              </td>
              <td
                data-testid="lastMonth"
              >
                7
              </td>
            </tr>
            <tr>
              <td>
                avg. per week
              </td>
              <td
                data-testid="avgWeek"
              >
                4.4
              </td>
            </tr>
            <tr>
              <td>
                avg. per month
              </td>
              <td
                data-testid="avgMonth"
              >
                5.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment after transactions are fetched with no transactions should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 150px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Consumption Statistics
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4 text-muted"
        >
          <div>
            Not Enough Data
          </div>
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                last week
              </td>
              <td
                data-testid="lastWeek"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                last month
              </td>
              <td
                data-testid="lastMonth"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                avg. per week
              </td>
              <td
                data-testid="avgWeek"
              >
                0
              </td>
            </tr>
            <tr>
              <td>
                avg. per month
              </td>
              <td
                data-testid="avgMonth"
              >
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Setup Environment before transactions are fetched with items in the inventory should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  border-style: solid;
  border-width: 1px;
  border-color: #000000;
}

.c1 {
  width: 100%;
  margin-left: 10px;
  margin-right: 10px;
  text-align: center;
}

.c2 {
  list-style-type: none;
  list-style-position: outside;
  margin: 0px;
  width: 100%;
  background: #808080;
  padding: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin: 5px;
  padding: 5px;
}

.c3 .table {
  margin-bottom: 0px;
}

.c3 .chartBox {
  height: 150px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 alert alert-success mb-2"
    >
      Consumption Statistics
    </div>
    <div
      class="c1 alert alert-warning mb-2"
    >
      1 Item(s) Expiring Soon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="chartBox"
        >
          <canvas
            id="consumptionChart"
          />
        </div>
      </div>
      <div
        class="c3"
      >
        <table
          class="table table-sm table-striped table-bordered table-hover"
        >
          <tbody>
            <tr>
              <td>
                current inventory
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                last week
              </td>
              <td
                data-testid="lastWeek"
              >
                6
              </td>
            </tr>
            <tr>
              <td>
                last month
              </td>
              <td
                data-testid="lastMonth"
              >
                7
              </td>
            </tr>
            <tr>
              <td>
                avg. per week
              </td>
              <td
                data-testid="avgWeek"
              >
                4.4
              </td>
            </tr>
            <tr>
              <td>
                avg. per month
              </td>
              <td
                data-testid="avgMonth"
              >
                5.5
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
